include/crab/domains/region_domain.hpp

```diff
@@ -561,6 +567,30 @@ class region_domain final
   base_abstract_domain_t m_base_dom;
   /** End base domain **/
 
+  /** Begin MRU domain **/
+  // The cache abstract domain that capture invariants for MRU object
+  base_abstract_domain_t m_cache_dom;
+  using base_variable_domain_t = variable_domain<base_variable_t>;
+  // left: the base address for cache
+  // right: a region variable to get equivalence class
+  using cache_variable_opt_t =
+      boost::optional<std::pair<base_variable_domain_t, variable_t>>;
+  cache_variable_opt_t m_cache_vars;
```

Comment:
I use this pair because we need to keep track of information for the cache domain. That is a reference for the MRU object and a region that MRU used.

Reply:
The solution is not good enough. Hope you give me some suggestions.